test/acquisition/test_analytic.py

```diff
@@ -819,96 +833,172 @@ def _get_model(self, dtype=torch.float):
             "covar_module.outputscale_prior.concentration": torch.tensor(2.0),
             "covar_module.outputscale_prior.rate": torch.tensor(0.1500),
         }
-        train_x = torch.linspace(0, 1, 10, device=self.device, dtype=dtype).unsqueeze(
-            -1
-        )
+        train_x = torch.linspace(
+            0.0, 1.0, 10, device=self.device, dtype=dtype
+        ).unsqueeze(-1)
+        # Taking the sin of the *transformed* input to make the test equivalent
+        # to when there are no input transforms
         train_y = torch.sin(train_x * (2 * math.pi))
+        # Now transform the input to be passed into SingleTaskGP constructor
+        train_x = train_x * (hi_x - low_x) + low_x
         noise = torch.tensor(NEI_NOISE, device=self.device, dtype=dtype)
         train_y += noise
         train_yvar = torch.full_like(train_y, 0.25**2)
-        model = SingleTaskGP(train_X=train_x, train_Y=train_y, train_Yvar=train_yvar)
-        model.load_state_dict(state_dict)
+        model = SingleTaskGP(
+            train_X=train_x,
+            train_Y=train_y,
+            train_Yvar=train_yvar,
+            outcome_transform=outcome_transform,
+            input_transform=input_transform,
+            covar_module=covar_module,
+        )
+        model.load_state_dict(state_dict, strict=False)
         model.to(train_x)
         model.eval()
         return model
 
     def test_noisy_expected_improvement(self):
-        for dtype in (torch.float, torch.double):
-            model = self._get_model(dtype=dtype)
-            X_observed = model.train_inputs[0]
-            nfan = 5
-            nEI = NoisyExpectedImprovement(model, X_observed, num_fantasies=nfan)
-            LogNEI = LogNoisyExpectedImprovement(model, X_observed, num_fantasies=nfan)
-            # before assigning, check that the attributes exist
-            self.assertTrue(hasattr(LogNEI, "model"))
-            self.assertTrue(hasattr(LogNEI, "best_f"))
-            self.assertIsInstance(LogNEI.model, SingleTaskGP)
-            self.assertIsInstance(LogNEI.model.likelihood, FixedNoiseGaussianLikelihood)
-            LogNEI.model = nEI.model  # let the two share their values and fantasies
-            LogNEI.best_f = nEI.best_f
-
-            X_test = torch.tensor(
-                [[[0.25]], [[0.75]]],
-                device=X_observed.device,
-                dtype=dtype,
+        # Same as the default Matern kernel
+        # botorch.models.utils.gpytorch_modules.get_matern_kernel_with_gamma_prior,
+        # except RBFKernel is used instead of MaternKernel.
+        # For some reason, RBF gives numerical problems but Matern does not.
```

Comment:
Which numerical problems did you run into? The RBF kernel is generally smoother as the Matern kernel, meaning its eigen-spectrum decays faster, which implies that associated covariance matrices are more likely to be numerically low rank than for the Matern. Evaluating the kernel on fewer points, or alternatively decreasing its lengthscale should make this type of numerical issue go away.

Reply:
You mean increasing the lengthscale right? But in any case, I think that would be too much of a pain because firstly, `NEI_NOISE` is fixed at 10 values. And secondly, the lengthscale really matters for making the tests work; whoever made this test I assume knew what they were doing and I don't want to arbitrarily fudge numbers (namely, the lengthscale) too much because I saw that that can make the tests fail. Why not just keep it how it is? 

How about this, if we really don't want those numerical warnings: test Matern on both float32 and float64, but only test RBF on float64?

Reply:
> You mean increasing the lengthscale right?

Increasing the lengthscale will tend to exacerbate the numerical ill conditioning of the RBF kernel matrix, decreasing the lengthscale will help it. You can think of the lengthscale as controlling the potential complexity of the function, and functions with larger lengthscales are smoother, i.e. less complex, than functions with shorter lengthscales, which can exhibit many more variations. This in turn shows up as an increase in the numerical rank for more complex functions, which helps the conditioning. 

In the most extreme case, a lengthscale of zero would imply that all points are independent, i.e. the covariance would be diagonal, and in the case of the unscaled RBF kernel, it'd be the identity, which has perfect conditioning.

> but only test RBF on float64

Sure!